src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -306,6 +311,9 @@ public CompilationTracker FreezePartialStateWithTree(SolutionState solution, Doc
                     // being made to the project, but it's the best we have so we'll use it.
                     inProgressCompilation = compilationWithoutGeneratedDocuments.AddSyntaxTrees(sourceGeneratedDocuments.Select(d => d.SyntaxTree));
 
+                    // This is likely a bug.  It seems possible to pass out a partial compilation state that we don't
+                    // properly record assembly symbols for.
```

Comment:
Hmm, is this really the same explanation as the comment you have later -- at this point the only remaining change this has to make is a source file change, which we're presuming won't change IAssemblySymbols for references. I'm not sure if we ever create InProgressStates where we didn't previously have a FinalState or not.

No request for a change here unless you think the comment can be revised.